### DISCUSSION TITLE
Fundamentos do Blazor com .NET 8 - Erro ao cadastrar novo produto

### DISCUSSION THREAD

**Discussion by gvlakatos** [balta-io/forum#475] [Tira Dúvidas] [answered]
Bom dia pessoal, tudo bem?

Ao tentar realizar a inclusão de um novo produto, a seguinte mensagem está aparecendo.

> The POST request does not specify which form is being submitted. To fix this, ensure <form> elements have a @formname attribute with any unique value, or pass a FormName parameter if using <EditForm>.

Não sei se há relação, mas precisei alterar o método de inicialização das categorias para o código abaixo, fazendo com que fique diferente do mostrado pelo Balta no curso (como demonstrado no curso, as categorias não estavam sendo carregadas no meu seletor).

```
protected override async Task OnInitializedAsync()
{
    Categories = await Context
                    .Categories
                    .AsNoTracking()
                    .ToListAsync();
}
```
No curso, esse método está assim.

```
protected override async void OnInitialized()
    {
        Categories = await Context
            .Categories
            .AsNoTracking()
            .OrderBy(x => x.Title)
            .ToListAsync();
    }
```

Alguém teria alguma ideia de como solucionar esse problema?

Obrigado!

**Comment by andrebaltieri** [comment]
> Bom dia pessoal, tudo bem?
> 
> 
> 
> Ao tentar realizar a inclusão de um novo produto, a seguinte mensagem está aparecendo.
> 
> 
> 
> > The POST request does not specify which form is being submitted. To fix this, ensure <form> elements have a @formname attribute with any unique value, or pass a FormName parameter if using <EditForm>.
> 
> 
> 
> Não sei se há relação, mas precisei alterar o método de inicialização das categorias para o código abaixo, fazendo com que fique diferente do mostrado pelo Balta no curso (como demonstrado no curso, as categorias não estavam sendo carregadas no meu seletor).
> 
> 
> 
> ```
> 
> protected override async Task OnInitializedAsync()
> 
> {
> 
>     Categories = await Context
> 
>                     .Categories
> 
>                     .AsNoTracking()
> 
>                     .ToListAsync();
> 
> }
> 
> ```
> 
> No curso, esse método está assim.
> 
> 
> 
> ```
> 
> protected override async void OnInitialized()
> 
>     {
> 
>         Categories = await Context
> 
>             .Categories
> 
>             .AsNoTracking()
> 
>             .OrderBy(x => x.Title)
> 
>             .ToListAsync();
> 
>     }
> 
> ```
> 
> 
> 
> Alguém teria alguma ideia de como solucionar esse problema?
> 
> 
> 
> Obrigado!

Boa tarde, @gvlakatos, como vai?

Se estiver usando o Postman, você precisa selecionar a opção RAW e em seguida no Content-Type marcar a opção Application/JSON!

Assim ele vai entender que é um post do tipo JSON e não um FORM.

**Reply by gvlakatos** [reply]
Olá @andrebaltieri ,

O erro acontece ao tentar criar um produto através da página Create dos Products.

![image](https://github.com/balta-io/forum/assets/37455296/c2bde60a-6f0c-4e69-a078-55bb07bd501c)

Analisando o Visual Studio, notei agora essa mensagem.

> Microsoft.Data.SqlClient.SqlException (0x80131904): The INSERT statement conflicted with the FOREIGN KEY constraint "FK_Products_Categories_CategoryId". The conflict occurred in database "BlazorShop", table "dbo.Categories", column 'Id'.

Abaixo está a foreign key da minha tabela Products.

![image](https://github.com/balta-io/forum/assets/37455296/0ec5499b-39db-4120-bd51-9bb643dce4d4)

E abaixo está a minha tabela Categories.

![image](https://github.com/balta-io/forum/assets/37455296/64d64fc1-671e-4879-acd1-e64e49020181)

**Reply by andrebaltieri** [reply]
@gvlakatos o ID vem preenchido? Se sim é um ID válido e que existe na tabela de categorias?

**Reply by gvlakatos** [reply, marked as answer]
Olá @andrebaltieri,

Por algum motivo, ao reiniciar o meu banco, dar um clean e build no meu projeto, a aplicação começou a funcionar corretamente.

Não sei se fiz algo de errado durante as etapas mas agora está funcionando tudo corretamente.

Obrigado @andrebaltieri !
